draft-yasskin-http-origin-signed-responses.md

```diff
@@ -501,26 +501,14 @@ to retrieve an updated OCSP from the original server.
          this returns "invalid", return "invalid".
       1. Let `main-certificate` be the first certificate in `certificate-chain`.
       1. Set `publicKey` to `main-certificate`'s public key.
-      1. The client MUST define a partial function from public key types to
-         signing algorithms, and this function must at the minimum include the
-         following mappings:
-
-         RSA, 2048 bits:
-         : rsa_pss_rsae_sha256 or rsa_pss_pss_sha256, as defined in Section
-           4.2.3 of {{!I-D.ietf-tls-tls13}}, depending on which of the
-           rsaEncryption OID or RSASSA-PSS OID {{!RFC8017}} is used.
-
-         EC, with the secp256r1 curve:
-         : ecdsa_secp256r1_sha256 as defined in Section 4.2.3 of
-           {{!I-D.ietf-tls-tls13}}.
-
-         EC, with the secp384r1 curve:
-         : ecdsa_secp384r1_sha384 as defined in Section 4.2.3 of
-           {{!I-D.ietf-tls-tls13}}.
-
-         Set `signing-alg` to the result of applying this function to the type of
-         `main-certificate`'s public key. If the function is undefined on this
-         input, return "invalid".
+      1. If `publicKey` is an RSA key, return "invalid".
+      1. If `publicKey` is a key using the secp256r1 elliptic curve, set
+         `signing-alg` to ecdsa_secp256r1_sha256 as defined in Section 4.2.3 of
+         {{!I-D.ietf-tls-tls13}}.
+      1. Otherwise, either return "invalid" or set `signing-alg` to a non-legacy
+         signing algorithm defined by TLS 1.3 or later
+         ({{!I-D.ietf-tls-tls13}}). This choice MUST depend only on
```

Comment:
Would it be possible to pin the TLS version/draft in the signedexchange spec?

Reply:
Do you mean to the -28 draft or to exactly TLS 1.3? The RFC is ["Submitted to IESG for Publication"](https://datatracker.ietf.org/doc/draft-ietf-tls-tls13/), so it's unlikely that -28 will change significantly before it's published as an RFC.

It's possible to pin to 1.3, but [Ilari's post](https://lists.w3.org/Archives/Public/ietf-http-wg/2018AprJun/0050.html) seems to say that he wants this RFC to allow for future expansion in the set of algorithms without needing to revise this RFC. Other security folks might speak up and disagree, of course.

Note that the WHATWG-side spec will be a living spec that'll follow the current version of TLS and specify exactly which algorithms browsers are using, even if this spec is less precise.